### PR TITLE
[Buttons] Add unit tests for Floating Button layout

### LINKED ISSUE #2570: [FloatingButton] Add unit test for multiple layout calls
### Overview
To verify that multiple layout passes will not introduce subtle insetting/repositioning errors, a unit test covering this behavior should be written.

### Reproduction steps
1. Layout a FAB 
1. Capture the frames of `imageView` and `titleLabel`
1. Force another layout of the same object
1. Verify the frames are identical 
1. Repeat for different shapes/modes/image alignments.

### Version number
40.1.1

### Related issues
PR Feedback from #2563

### PR DESCRIPTION
**Pull request by romoore**
A few more unit tests to verify the layout behavior of MDCFloatingButton.

Closes #2570

### CODE REVIEW

File: components/Buttons/tests/unit/FloatingButtonTests.m [hunk 1]
```diff
@@ -488,13 +498,38 @@ - (void)testExpandedLayout {
...
+                  inMode:MDCFloatingButtonModeExpanded];
+  button.imageTitleSpacing = 12;
+
+  // When
+  [button sizeToFit];
+
+  // Then
+  CGRect imageFrame = button.imageView.frame;
+  CGRect titleFrame = button.titleLabel.frame;
+  XCTAssertEqualWithAccuracy(CGRectGetMaxX(imageFrame), CGRectGetWidth(button.bounds)-16, 1);
```
Comment (jverkoey) [inline components/Buttons/tests/unit/FloatingButtonTests.m:520]: Where does the -16 come from?

File: components/Buttons/tests/unit/FloatingButtonTests.m [hunk 2]
```diff
@@ -518,6 +553,33 @@ - (void)testExpandedLayoutWithNonZeroContentEdgeInsets {
...
+  [button layoutIfNeeded];
+
+  // Then
+  XCTAssertTrue(CGRectEqualToRect(originalImageFrame, button.imageView.frame),
+                @"Image frames are not equal.\nExpected: %@\nReceived: %@",
+                NSStringFromCGRect(originalImageFrame), NSStringFromCGRect(button.imageView.frame));
+  XCTAssertTrue(CGRectEqualToRect(originalTitleFrame, button.titleLabel.frame),
+                @"Title frames are not equal.\nExpected: %@\nReceived: %@",
+                NSStringFromCGRect(originalTitleFrame),
+                NSStringFromCGRect(button.titleLabel.frame));
```
Comment (jverkoey) [inline components/Buttons/tests/unit/FloatingButtonTests.m:579]: 👍

**Review by jverkoey — APPROVED** [review]
LGTM with minor question about constants.

File: components/Buttons/tests/unit/FloatingButtonTests.m (hunk 1 above)
Comment (romoore) [reply components/Buttons/tests/unit/FloatingButtonTests.m:520]: It's from `internalLayoutSpacing.left` but in this case flipped for the `.trailing` imageLocation value.  It's not great that it's both here and in the implementation, but I didn't feel like it was worth externing the UIEdgeInsets in the implementation.

**Comment by romoore** [comment]
Travis failed.

**Comment by romoore** [comment]
```
/Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:512:10: error: property 'imageTitleSpacing' not found on object of type 'MDCFloatingButton *'; did you mean 'imageTitleSpace'?
  button.imageTitleSpacing = 12;
         ^~~~~~~~~~~~~~~~~
         imageTitleSpace
In module 'MaterialComponents' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:19:
/Users/travis/build/material-components/material-components-ios/components/Buttons/src/MDCFloatingButton.h:96:38: note: 'imageTitleSpace' declared here
@property(nonatomic, assign) CGFloat imageTitleSpace UI_APPEARANCE_SELECTOR;
                                     ^
/Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:523:65: error: property 'imageTitleSpacing' not found on object of type 'MDCFloatingButton *'; did you mean 'imageTitleSpace'?
                             CGRectGetMaxX(titleFrame) + button.imageTitleSpacing, 1);
                                                                ^~~~~~~~~~~~~~~~~
                                                                imageTitleSpace
In module 'XCTest' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:17:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/Headers/XCTestAssertions.h:136:76: note: expanded from macro 'XCTAssertEqualWithAccuracy'
    _XCTPrimitiveAssertEqualWithAccuracy(self, expression1, @#expression1, expression2, @#expression2, accuracy, @#accuracy, __VA_ARGS__)
                                                                           ^
In module 'XCTest' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:17:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/Headers/XCTestAssertionsImpl.h:251:20: note: expanded from macro '_XCTPrimitiveAssertEqualWithAccuracy'
        __typeof__(expression2) expressionValue2 = (expression2); \
                   ^
In module 'MaterialComponents' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:19:
/Users/travis/build/material-components/material-components-ios/components/Buttons/src/MDCFloatingButton.h:96:38: note: 'imageTitleSpace' declared here
@property(nonatomic, assign) CGFloat imageTitleSpace UI_APPEARANCE_SELECTOR;
                                     ^
/Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:523:65: error: property 'imageTitleSpacing' not found on object of type 'MDCFloatingButton *'; did you mean 'imageTitleSpace'?
                             CGRectGetMaxX(titleFrame) + button.imageTitleSpacing, 1);
                                                                ^~~~~~~~~~~~~~~~~
                                                                imageTitleSpace
In module 'XCTest' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:17:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/Headers/XCTestAssertions.h:136:76: note: expanded from macro 'XCTAssertEqualWithAccuracy'
    _XCTPrimitiveAssertEqualWithAccuracy(self, expression1, @#expression1, expression2, @#expression2, accuracy, @#accuracy, __VA_ARGS__)
                                                                           ^
In module 'XCTest' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:17:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/Headers/XCTestAssertionsImpl.h:251:53: note: expanded from macro '_XCTPrimitiveAssertEqualWithAccuracy'
        __typeof__(expression2) expressionValue2 = (expression2); \
                                                    ^
In module 'MaterialComponents' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:19:
/Users/travis/build/material-components/material-components-ios/components/Buttons/src/MDCFloatingButton.h:96:38: note: 'imageTitleSpace' declared here
@property(nonatomic, assign) CGFloat imageTitleSpace UI_APPEARANCE_SELECTOR;
                                     ^
/Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:523:65: error: property 'imageTitleSpacing' not found on object of type 'MDCFloatingButton *'; did you mean 'imageTitleSpace'?
                             CGRectGetMaxX(titleFrame) + button.imageTitleSpacing, 1);
                                                                ^~~~~~~~~~~~~~~~~
                                                                imageTitleSpace
In module 'XCTest' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:17:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/Headers/XCTestAssertions.h:136:76: note: expanded from macro 'XCTAssertEqualWithAccuracy'
    _XCTPrimitiveAssertEqualWithAccuracy(self, expression1, @#expression1, expression2, @#expression2, accuracy, @#accuracy, __VA_ARGS__)
                                                                           ^
In module 'XCTest' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:17:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/Headers/XCTestAssertionsImpl.h:255:104: note: expanded from macro '_XCTPrimitiveAssertEqualWithAccuracy'
            NSValue *expressionBox2 = [NSValue value:&expressionValue2 withObjCType:@encode(__typeof__(expression2))]; \
                                                                                                       ^
In module 'MaterialComponents' imported from /Users/travis/build/material-components/material-components-ios/components/Buttons/tests/unit/FloatingButtonTests.m:19:
/Users/travis/build/material-components/material-components-ios/components/Buttons/src/MDCFloatingButton.h:96:38: note: 'imageTitleSpace' declared here
@property(nonatomic, assign) CGFloat imageTitleSpace UI_APPEARANCE_SELECTOR;
                                     ^
4 errors generated.
```

**Pull request merged**